### PR TITLE
Use gcr.io/distroless/base for the build-base image

### PR DESCRIPTION
**Pull request by dprotaso**
This makes Knative container images uniform for the sake of
license scanning

Bazel is introduced again so we can use rules provided by
github.com/GoogleContainerTools/distroless to build our base
container images

### CODE REVIEW

File: images/BUILD [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+container_image(
+    name = "build-base-debug",
+    base = "@base-debug//image",
+    debs = [
+        packages["openssh-client"],
+        packages["libedit2"],
+        packages["libgssapi-krb5-2"],
+        packages["libselinux1"],
+        packages["libssl1.0.2"],
+        #packages["zlib1g"],
```
Comment (ImJasonH) [inline images/BUILD]: Remove this?

File: images/BUILD [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        packages["openssh-client"],
+        packages["git"],
+    ],
+)
+
+container_image(
+    name = "build-base-debug",
+    base = "@base-debug//image",
+    debs = [
+        packages["openssh-client"],
```
Comment (ImJasonH) [inline images/BUILD]: Are packages like `libedit2` necessary for `openssh-client`? Can you add a comment to that effect?

Comment (dprotaso) [reply images/BUILD]: There's a comment a link to the package in `WORKSPACE` that shows required deps

Comment (dprotaso) [reply images/BUILD]: I'll mention go to `WORKSPACE` for comments

**Comment by dprotaso** [comment]
/hold cancel

@ImJasonH let me know if there's anything else
@adrcunha is there a way to test the release script?

File: hack/release.sh [hunk 3]
```diff
@@ -74,3 +89,6 @@ publish_yaml ${OUTPUT_YAML} ${BUILD_RELEASE_GCS} ${TAG}
 echo "New release published successfully"
 
 # TODO(mattmoor): Create other aliases?
+
+# Cleanup bazel artifacts
+bazel clean --expunge
```
Comment (adrcunha) [inline hack/release.sh]: Better trap EXIT and run the cleanup?

File: hack/release.sh [hunk 4]
```diff
@@ -40,7 +41,21 @@ run_validation_tests ./test/presubmit-tests.sh
...
+  --define repository=${BUILD_BASE_REPO} \
+  //images:all
+
+bazel run \
+  --define registry=${BUILD_BASE_REGISTRY} \
+  --define repository=${BUILD_BASE_REPO} \
+  //images:push-build-base
+
+bazel run \
+  --define registry=gcr.io \
```
Comment (adrcunha) [inline hack/release.sh]: `s/gcr.io/${BUILD_BASE_REGISTRY}`?

**Comment by adrcunha** [comment]
@dprotaso you can simply run the release script locally to build a new nightly release. You can use `--nopublish` to avoid it being pushed to the GCR, if you want.

File: hack/release.sh [hunk 5]
```diff
@@ -23,7 +23,8 @@ readonly BUILD_RELEASE_GCS
 readonly BUILD_RELEASE_GCR
 
 # Location of the base image for creds-init and git images
-readonly BUILD_BASE_GCR="${BUILD_RELEASE_GCR}/github.com/knative/build/build-base"
+readonly BUILD_BASE_REGISTRY="gcr.io"
```
Comment (adrcunha) [inline hack/release.sh]: You should change this to `ko.local` is `PUBLISH` release is 1.

Comment (adrcunha) [reply hack/release.sh]:
```suggestion
(( PUBLISH_RELEASE )) && BUILD_BASE_REGISTRY="gcr.io" || BUILD_BASE_REGISTRY="ko.local"
```

**Comment by dprotaso** [comment]
/hold 
running the full script locally

File: hack/release.sh [hunk 6]
```diff
@@ -23,24 +23,43 @@ readonly BUILD_RELEASE_GCS
...
 
 parse_flags $@
 
 set -o errexit
 set -o pipefail
 
-run_validation_tests ./test/presubmit-tests.sh
+trap bazel_cleanup EXIT
+
+#run_validation_tests ./test/presubmit-tests.sh
```
Comment (adrcunha) [inline hack/release.sh]: Uncomment. Use `--skip-tests` to skip tests for debugging.

File: hack/release.sh [hunk 7]
```diff
@@ -49,24 +68,28 @@ export K8S_CLUSTER_OVERRIDE=CLUSTER_NOT_SET
...
-echo "New release built successfully"
+bazel run \
+  --define registry=${BUILD_BASE_REGISTRY} \
+  --define repository=${BUILD_BASE_REPO} \
+  //images:push-build-base-debug
 
-if (( ! PUBLISH_RELEASE )); then
- exit 0
-fi
+echo "- Destination GCR: ${BUILD_RELEASE_GCR}"
```
Comment (adrcunha) [inline hack/release.sh]: Move these to line 76, before the first push command.

**Pull request merged**